src/daft-session/src/session.rs

```diff
@@ -59,39 +54,58 @@ impl Session {
     }
 
     /// Attaches a catalog to this session, err if already exists.
-    pub fn attach(&self, name: String, catalog: Arc<dyn DataCatalog>) -> Result<()> {
-        if self.state().catalogs.exists(&name) {
-            obj_already_exists_err!("Catalog", &name.into())
+    pub fn attach_catalog(&self, catalog: CatalogRef, alias: String) -> Result<()> {
+        if self.state().catalogs.exists(&alias) {
+            obj_already_exists_err!("Catalog", &alias.into())
         }
-        self.state_mut().catalogs.attach(name, catalog);
+        self.state_mut().catalogs.insert(alias, catalog);
         Ok(())
     }
 
-    /// Detaches a catalog from this session, err if does not exist.
-    pub fn detach(&self, name: &str) -> Result<()> {
-        if self.state().catalogs.exists(name) {
-            obj_not_found_err!("Catalog", &name.into())
+    /// Attaches a table to this session, err if already exists.
+    pub fn attach_table(&self, table: TableRef, alias: impl Into<String>) -> Result<()> {
```

Comment:
nit(opinion): i think it feels more natural to do the name first. 

```py
sess.attach_catalog("my_catalog", catalog)
sess.attach_table("name", tbl)
```

Reply:
I believe the UX we wanted is for names to be optional hence placement.